### PR TITLE
Resolver: use new methods to resolve documentation pages

### PR DESCRIPTION
**Pull request by stsewd**
This may result in one queryset less in some cases, others will remain the same.

### CODE REVIEW

File: readthedocs/proxito/views/serve.py [hunk 1]
```diff
@@ -912,14 +912,15 @@ def changefreqs_generator():
 
             if project.translations.exists():
                 for translation in project.translations.all():
-                    translation_versions = (
```
Comment (stsewd) [inline readthedocs/proxito/views/serve.py:915]: We were fetching all versions from the translation, when we just need to get a version matching the same slug as the current version.

File: readthedocs/embed/views.py [hunk 2]
```diff
@@ -135,12 +134,10 @@ def get(self, request):
 def do_embed(*, project, version, doc=None, path=None, section=None, url=None):
     """Get the embed response from a document section."""
     if not url:
-        external = version.type == EXTERNAL
```
Comment (ericholscher) [inline readthedocs/embed/views.py:138]: Do we not need this anymore?

File: readthedocs/projects/models.py [hunk 3]
```diff
@@ -112,7 +112,7 @@ def save(self, *args, **kwargs):
 
     # HACK
     def get_absolute_url(self):
-        return resolve(self.child)
+        return resolver.resolve_version(self.child)
```
Comment (ericholscher) [inline readthedocs/projects/models.py:115]: Would be useful to pass a kwarg here. I thought we were only using resolve_version when we had a version object?

**Review by ericholscher — APPROVED** [review]

File: readthedocs/embed/views.py (hunk 2 above)
Comment (stsewd) [reply readthedocs/embed/views.py:138]: We get the type of the version from the version object itself

**Pull request merged**